core/jazz_ui/src/app/pages/environment-overview/env-overview-section.component.ts

```diff
@@ -251,9 +268,163 @@ popup(state){
   }
   openSidebar(){
     this.open_sidebar.emit(true);
-
 }
 
+  getSafeDetails() {
+    this.reqArnArray = [];
+    this.noData = false;
+    this.rolesList = [];
+    this.http.get(`/jazz/t-vault/safes/${this.safeName}`).subscribe(
+      (res) => {
+        if(Object.keys(res.data.roles).length === 0){
+          this.noData = true;
+          this.isSecretLoading = false;
+        }
+        else {
+        this.rolesObj = res.data.roles;
+        this.isSecretLoading = false;
+        this.rolesList.push(Object.keys(this.rolesObj))
+        this.rolesList[0].map((item)=>{
+            this.reqArnArray.push(this.rolesObj[item].arn)
+        })
+        if(this.reqArnArray.length > 1){
+          this.firstSafeRole = this.reqArnArray.splice(0,1)
+        } else  if(this.reqArnArray.length === 1){
+          this.firstSafeRole = this.reqArnArray[0];
+          this.reqArnArray = []
+        }
+       }
+      },
+      (error) => {
+        this.isError = true;
+        this.isSecretLoading = false;
+        let errMessage = this.toastmessage.errorMessage(error, 'updateSecret');
+        this.toast_pop('error', 'Oops!', errMessage);
+      }
+    )
+  }
+ 
+  addRoleInSafe() {
+    this.isSecretLoading = true;
+    let payload = {
+      "arn": this.roleValue,
+      "permission": "read"
+    }
+    this.http.post(`/jazz/t-vault/safes/${this.safeName}/role`, payload).subscribe(
+      res => {
+        this.isAddOrDelete = false;
+        let successMessage = this.toastmessage.successMessage(res,"updateSecret");
+        this.toast_pop('success',"",successMessage);
```

Comment:
Please delimit with single quotes or the dominant standard.

Reply:
done

Reply:
done